client/src/App.js

```diff
@@ -13,23 +16,41 @@ const Container = styled.div`
   height: 100%;
   min-height: 100vh;
 `
+export const UserContext = createContext([null, () => {}])
 
-const App = () => (
-  <Container>
-    <Reset />
-    <Global />
-    <BrowserRouter>
-      <Route path="/" exact>
-        <Landing />
-      </Route>
-      <Route path="/conversationTest">
-        <Conversation />
-      </Route>
+function App() {
+  const [user, setUser] = useState(null)
 
-      <Route path="/login" render={() => <h1>We're in signin</h1>} />
-      <Route path="/register" render={() => <h1>We're in signup</h1>} />
-    </BrowserRouter>
-  </Container>
-)
+  useEffect(() => {
+    async function authWithCookie() {
+      const { user } = await authApi.getCachedUser()
```

Comment:
if any validation problem comes up, the api service will be the one to take care of it since it handles token management. In our case, it just returns a resolved promise with a value of `{ user: null}`. This is a design decision and would likely get expanded in the future as we decide to add to the service.
The main idea behind the design was separation of concerns: the api also takes care of token errors while UI errors are delegated to the UI components